mne/io/nirx/nirx.py

```diff
@@ -87,10 +88,28 @@ def __init__(self, fname, saturated, preload=False, verbose=None):
 
         fname = _check_fname(fname, 'read', True, 'fname', need_dir=True)
 
+        json_config = glob.glob('%s/*%s' % (fname, "config.json"))
+        if len(json_config):
+            is_aurora = True
+        else:
+            is_aurora = False
+
+        if is_aurora:
+            # NIRSport2 devices using Aurora software
+            keys = ('hdr', 'config.json', 'description.json',
+                    'wl1', 'wl2', 'probeInfo.mat', 'tri')
+        else:
+            # NIRScout devices and NIRSport1 devices
+            keys = ('hdr', 'inf', 'set', 'tpl', 'wl1', 'wl2',
+                    'config.txt', 'probeInfo.mat')
+            if len(glob.glob('%s/*%s' % (fname, 'dat'))) != 1:
+                warn("A single dat file was expected in the specified path, "
+                     "but got %d. This may indicate that the file structure "
+                     "has been modified since the measurement was saved." %
+                     (len(glob.glob('%s/*%s' % (fname, 'dat')))))
```

Comment:
DRY is nice here because in the future if we need to improve the code for filename detection we only have to do it in one place. Also we can use f-strings nowadays
```suggestion
            n_dat = len(glob.glob(f'{fname}/{dat}'))
            if n_dat != 1:
                warn("A single dat file was expected in the specified path, "
                     f"but got {n_dat}. This may indicate that the file "
                     "structure has been modified since the measurement "
                     "was saved.")
```